packages/edit-site/src/components/add-new-template/new-template-part.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * External dependencies
+ */
+import { kebabCase } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+import { Button } from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import CreateTemplatePartModal from '../create-template-part-modal';
+
+export default function NewTemplatePart( { postType } ) {
+	const [ isModalOpen, setIsModalOpen ] = useState( false );
+	const { saveEntityRecord } = useDispatch( coreStore );
+
+	async function createTemplatePart( { title, area } ) {
+		if ( ! title || ! area ) {
+			return;
+		}
+
+		const templatePart = await saveEntityRecord(
+			'postType',
+			'wp_template_part',
+			{
+				slug: kebabCase( title ),
+				title,
+				content: '',
+				area,
+			}
+		);
```

Comment:
It is a bit glitchy right now with the way the entity is added in the background, and then the page redirects to the editor.

Possibly a work around is to use `apiFetch` to create the entity so that the core data store isn't updated when doing this. We navigate away anyway, which destroys the store state.